modules/standard/Set.chpl

```diff
@@ -784,6 +799,15 @@ module Set {
     return result;
   }
 
+  pragma "no doc"
+  operator :(x: set(?et1, ?p1), type t: set(?et2, ?p2)) {
+    if et1 != et2 then
+      compilerError('Cannot cast to set with different ',
+                    'element type: ', t:string);
+    var result: set(et1, p2) = x;
+    return result;
+  }
```

Comment:
* I propose to be proud and un-nodoc this cast.

* Add  a TODO comment suggesting to allow subtyping in element types, once `init=` supports that.

Reply:
I don't really want to broadcast that we have it because I'm not even sure that we _want_ to have it.

I had to add it because the compiler complains about having `init=`, `=`, but not a cast.

Conceptually speaking, having a user copy a set just to cast off the `parSafe` value seems like a huge waste. I'd be inclined to revisit / make it public after we talk about coercion of collection elements some more.

Reply:
Reading through this again, I should have put a comment in explaining why I introduced the cast in the first place. Mistake on my part.